docs/data/data-grid/row-grouping/row-grouping.md

```diff
@@ -191,6 +191,14 @@ In the example below, the `director` column can not be grouped. And in all examp
 
 {{"demo": "RowGroupingColDefCanBeGrouped.js", "bg": "inline", "defaultCodeOpen": false}}
 
+### Grouping non-groupable columns programmatically
+
+The columns with `colDef.groupable` set to `false` could be grouped programmatically by initialiing, passing in the `rowGroupingModel` prop, or updating `rowGroupingModel` by the API method `setRowGroupingModel` to generate read-only row groups. The users could not alter such groups from the UI.
```

Comment:
What do you think about structuring it similarly to https://github.com/mui/mui-x/pull/11574:

<img width="822" src="https://github.com/mui/mui-x/assets/13808724/7183befa-8c4c-4e58-b79e-cb295064045a">

Reply:
Nice idea 💡